feat: add vm networking at schedule (#15)
polyxia-org/rik: riklet/README.md

## Usage

### Général Usage

**To run the Riklet, you must have setup previously your RIK master node and get
it's IP address.**

```bash
riklet --master-ip <YOUR_IP>:<PORT>
```

To run riklet as root with **cargo**:

```bash
CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_RUNNER='sudo -E' cargo run --bin riklet
```

### Faas Usage

**Prerequisite**: You need firecracker in yout PATH.

To run riklet with FAAS configuration.

```bash

> sudo riklet --firecracker-path <FIRECRACKER_LOCATION> --kernel-path <KERNEL_LOCATION> --ifnet <IFACE> --ifnet-ip <IFACE_IP> --script-path <SCRIPT_LOCATION>

Reviewer: Please consider put the table of variable needed here, and then you can refer them with `${NAME}` in your example command. 

That way it might be clearer to determine that variables are needed at first, then user is able to run

```suggestion
sudo riklet --firecracker-path ${FIRECRACKER_LOCATION} \
    --kernel-path ${KERNEL_LOCATION} --ifnet ${IFACE} \
    --ifnet-ip ${IFACE_IP} --script-path ${SCRIPT_LOCATION}
```

```

Exemple:

```bash
sudo riklet --firecracker-path $(which firecracker) --kernel-path ./vmlinux.bin --ifnet wlp2s0 --ifnet-ip 192.168.1.84 --script-path ./scripts/setup-host-tap.sh
```

> 
> `IFACE`: Network interface connected to the internet.
> 
> `IFACE_IP`: IP of the Network interface connected to the internet.
> 
> `FIRECRACKER_LOCATION`: Path to the firecracker binary.
> 
> `KERNEL_LOCATION`: Path to the kernel location.
> 
> `SCRIPT_LOCATION`: Path to the
> [script](https://github.com/polyxia-org/rik/blob/main/scripts/setup-host-tap.sh)
> that create tap interface.
> 
> You can set this configuration as environement variable.

Reviewer: A table seems to be more appropriate here

```suggestion

Here is a list of **required** environment variables to run riklet:

| Environment Variable | Description | Default |
|---|---|---|
| `IFACE` | Network interface connected to the internet | "" |
| `IFACE_IP` | IP of the Network interface connected to the internet | "" |
| `FIRECRACKER_LOCATION` | Path to the firecracker binary | "" |
| `KERNEL_LOCATION` | Path to the kernel location | "" |
| `SCRIPT_LOCATION` | Path to the
[script](https://github.com/polyxia-org/rik/blob/main/scripts/setup-host-tap.sh)
that create tap interface. | "" |
```

[2021-07-03T15:04:09Z INFO  riklet::core] Riklet (v0.1.0) is ready to accept connections.
```

## Cleaning up

> For now the Riklet don't clean his network configuration by his self.

Reviewer: ```suggestion
For now the Riklet doesn't clean its network configuration by itself.
```

To clean the network configuration:

```bash
sudo iptables -D FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT

> sudo iptables -D FORWARD -i rik-<FUNCTION_NAME>-tap -o <IFACE> -j ACCEPT

Reviewer: ```suggestion
sudo iptables -D FORWARD -i rik-${FUNCTION_NAME}-tap -o ${IFACE} -j ACCEPT
```

> sudo iptables -t nat -D POSTROUTING -o <IFACE> -j MASQUERADE

Reviewer: ```suggestion
sudo iptables -t nat -D POSTROUTING -o ${IFACE} -j MASQUERADE
```

```

> `FUNCTION_NAME`: the name of the function that is scheduled.

Reviewer: Please move this to a table above the shell example
